### PR TITLE
NMS-12358: Added BMP parser

### PR DESCRIPTION
**Pull request by fooker**
This adds a telemtryd parser for the BMP protocol.

* JIRA (Issue Tracker): http://issues.opennms.org/browse/NMS-12385

### CODE REVIEW

File: features/telemetry/protocols/bmp/parser/src/main/resources/blueprint/blueprint.xml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+		http://aries.apache.org/schemas/blueprint-cm/blueprint-cm-1.1.0.xsd
+		http://aries.apache.org/blueprint/xmlns/blueprint-ext/v1.1.0
+		http://aries.apache.org/schemas/blueprint-ext/blueprint-ext-1.1.xsd
+">
+    <reference id="telemetryRegistry" interface="org.opennms.netmgt.telemetry.api.registry.TelemetryRegistry" availability="mandatory" />
+<!--    <reference id="eventForwarder" interface="org.opennms.netmgt.events.api.EventForwarder" availability="mandatory"/>-->
+<!--    <reference id="identity" interface="org.opennms.distributed.core.api.Identity" />-->
+<!--    <reference id="dnsResolver" interface="org.opennms.netmgt.dnsresolver.api.DnsResolver" availability="mandatory" />-->
+
+    <!-- Netflow 5 UDP Parser Factory -->
```
Comment (j-white) [inline features/telemetry/protocols/bmp/parser/src/main/resources/blueprint/blueprint.xml]: Netflow vs BMP

File: features/telemetry/protocols/bmp/parser/src/main/resources/blueprint/blueprint.xml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+           xsi:schemaLocation="
+		http://www.osgi.org/xmlns/blueprint/v1.0.0
+		http://www.osgi.org/xmlns/blueprint/v1.0.0/blueprint.xsd
+		http://aries.apache.org/blueprint/xmlns/blueprint-cm/v1.1.0
+		http://aries.apache.org/schemas/blueprint-cm/blueprint-cm-1.1.0.xsd
+		http://aries.apache.org/blueprint/xmlns/blueprint-ext/v1.1.0
+		http://aries.apache.org/schemas/blueprint-ext/blueprint-ext-1.1.xsd
+">
+    <reference id="telemetryRegistry" interface="org.opennms.netmgt.telemetry.api.registry.TelemetryRegistry" availability="mandatory" />
+<!--    <reference id="eventForwarder" interface="org.opennms.netmgt.events.api.EventForwarder" availability="mandatory"/>-->
```
Comment (j-white) [inline features/telemetry/protocols/bmp/parser/src/main/resources/blueprint/blueprint.xml]: Remove commented references.

File: features/telemetry/protocols/bmp/parser/src/main/java/org/opennms/netmgt/telemetry/protocols/bmp/parser/proto/bgp/Packet.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+import org.opennms.netmgt.telemetry.protocols.bmp.parser.InvalidPacketException;
+import org.opennms.netmgt.telemetry.protocols.bmp.parser.proto.bgp.packets.KeepalivePacket;
+import org.opennms.netmgt.telemetry.protocols.bmp.parser.proto.bgp.packets.NotificationPacket;
+import org.opennms.netmgt.telemetry.protocols.bmp.parser.proto.bgp.packets.OpenPacket;
+import org.opennms.netmgt.telemetry.protocols.bmp.parser.proto.bgp.packets.UpdatePacket;
+import org.opennms.netmgt.telemetry.protocols.bmp.parser.proto.bmp.PeerFlags;
+
+public interface Packet {
+    void visit(final Visitor visitor);
```
Comment (mvrueden) [inline features/telemetry/protocols/bmp/parser/src/main/java/org/opennms/netmgt/telemetry/protocols/bmp/parser/proto/bgp/Packet.java]: This should be `accept(Visitor)` instead

File: features/telemetry/protocols/bmp/parser/src/main/java/org/opennms/netmgt/telemetry/protocols/bmp/parser/proto/bgp/packets/pathattr/Attribute.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.netmgt.telemetry.protocols.bmp.parser.proto.bgp.packets.pathattr;
+
+public interface Attribute {
+    void visit(final Visitor visitor);
```
Comment (mvrueden) [inline features/telemetry/protocols/bmp/parser/src/main/java/org/opennms/netmgt/telemetry/protocols/bmp/parser/proto/bgp/packets/pathattr/Attribute.java]: This should be `accept(Visitor)` instead

File: features/telemetry/protocols/bmp/parser/src/main/java/org/opennms/netmgt/telemetry/protocols/bmp/parser/proto/bmp/Packet.java [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+import org.opennms.netmgt.telemetry.protocols.bmp.parser.proto.bmp.packets.InitiationPacket;
+import org.opennms.netmgt.telemetry.protocols.bmp.parser.proto.bmp.packets.PeerDownPacket;
+import org.opennms.netmgt.telemetry.protocols.bmp.parser.proto.bmp.packets.PeerUpPacket;
+import org.opennms.netmgt.telemetry.protocols.bmp.parser.proto.bmp.packets.RouteMirroringPacket;
+import org.opennms.netmgt.telemetry.protocols.bmp.parser.proto.bmp.packets.RouteMonitoringPacket;
+import org.opennms.netmgt.telemetry.protocols.bmp.parser.proto.bmp.packets.StatisticsReportPacket;
+import org.opennms.netmgt.telemetry.protocols.bmp.parser.proto.bmp.packets.TerminationPacket;
+
+public interface Packet {
+    void visit(final Visitor visitor);
```
Comment (mvrueden) [inline features/telemetry/protocols/bmp/parser/src/main/java/org/opennms/netmgt/telemetry/protocols/bmp/parser/proto/bmp/Packet.java]: This should be `accept(Visitor)` instead

File: features/telemetry/protocols/bmp/parser/src/main/java/org/opennms/netmgt/telemetry/protocols/bmp/parser/proto/bmp/packets/mirroring/Mirroring.java [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.netmgt.telemetry.protocols.bmp.parser.proto.bmp.packets.mirroring;
+
+public interface Mirroring {
+    void visit(final Visitor visitor);
```
Comment (mvrueden) [inline features/telemetry/protocols/bmp/parser/src/main/java/org/opennms/netmgt/telemetry/protocols/bmp/parser/proto/bmp/packets/mirroring/Mirroring.java]: This should be `accept(Visitor)` instead

File: features/telemetry/protocols/bmp/parser/src/main/java/org/opennms/netmgt/telemetry/protocols/bmp/parser/proto/bmp/packets/down/Reason.java [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.netmgt.telemetry.protocols.bmp.parser.proto.bmp.packets.down;
+
+public interface Reason {
+    void visit(final Visitor visitor);
```
Comment (mvrueden) [inline features/telemetry/protocols/bmp/parser/src/main/java/org/opennms/netmgt/telemetry/protocols/bmp/parser/proto/bmp/packets/down/Reason.java]: This should be `accept(Visitor)` instead

File: features/telemetry/protocols/bmp/parser/src/main/java/org/opennms/netmgt/telemetry/protocols/bmp/parser/proto/bmp/packets/stats/Metric.java [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.netmgt.telemetry.protocols.bmp.parser.proto.bmp.packets.stats;
+
+public interface Metric {
+    void visit(final Visitor visitor);
```
Comment (mvrueden) [inline features/telemetry/protocols/bmp/parser/src/main/java/org/opennms/netmgt/telemetry/protocols/bmp/parser/proto/bmp/packets/stats/Metric.java]: This should be `accept(Visitor)` instead

**Pull request merged**